pkg/framework/objects/service_token/data_source.go

```diff
@@ -0,0 +1,138 @@
+package service_token
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/dbt-labs/terraform-provider-dbtcloud/pkg/dbt_cloud"
+	"github.com/dbt-labs/terraform-provider-dbtcloud/pkg/helper"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ datasource.DataSource              = &serviceTokenDataSource{}
+	_ datasource.DataSourceWithConfigure = &serviceTokenDataSource{}
+)
+
+func ServiceTokenDataSource() datasource.DataSource {
+	return &serviceTokenDataSource{}
+}
+
+type serviceTokenDataSource struct {
+	client *dbt_cloud.Client
+}
+
+// Metadata implements datasource.DataSource.
+func (st *serviceTokenDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_token"
+}
+
+// Configure implements datasource.DataSourceWithConfigure.
+func (st *serviceTokenDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	switch c := req.ProviderData.(type) {
+	case nil: // do nothing
+	case *dbt_cloud.Client:
+		st.client = c
+	default:
+		resp.Diagnostics.AddError("Missing client", "A client is required to configure the service token resource")
+	}
+}
+
+// Schema implements datasource.DataSource.
+func (st *serviceTokenDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"service_token_id": schema.Int64Attribute{
+				Required:    true,
+				Description: "The ID of the service token",
+				// TODO(cwalden): Deprecate this in favor of `id`
+				// DeprecationMessage: "Use `id` instead",
```

Comment:
Yes, in the SDKv2, `id` was a "special" field, a string that was supposed to hold the ID of the resource. With the Framework we don't really need it and might think of whether we want to deprecate the field for all resources/datasources once everything has moved to the Framework.

I also don't think that right now we are consistent with what the resource ID is (whether `id` or `<resource_type>_id`. Not a huge deal, but not ideal.

Reply:
IMO, I'd prefer `id` over `<resource_type>_id`. It better aligns with the APIs. Based on other usage patterns in the APIs and Provider, `<resource_type>_id` implies a "foreign key" style relationship, while `id` is "primary key"

Reply:
Lets have a chat on this. There are a few tricky things to contend with here.
1. SDKv2's built-in `id` is always a String, but we typically use Int64. Do we stick with Strings and always convert to/from Int64? Changing it is a breaking change so any user that relies on it being a String could face some issues.
2. How do we handle the removal of `service_token_id`?
3. Will we need/want to implement a StateUpgrader?